.travis.yml

```diff
@@ -1,33 +1,66 @@
+sudo: required
+
 language:
   - cpp
-#  - objective-c
 
-dist: trusty
-sudo: required
+os:
+  - linux
+  - osx
+
+cache: ccache
 
 compiler:
   - gcc
   - clang
 
-install:
-  - sudo add-apt-repository ppa:zoogie/sdl2-snapshots -y
-  - sudo add-apt-repository ppa:george-edison55/cmake-3.x -y
-  - sudo apt-get -qq update
-  - sudo apt-get -qq install libwebp-dev libglew-dev libgmp-dev libjpeg8-dev libcurl4-gnutls-dev libfreetype6-dev libgeoip-dev libncursesw5-dev libogg-dev libopenal-dev libpng-dev libsdl2-dev libtheora-dev libvorbis-dev libopusfile-dev libxvidcore-dev nettle-dev zlib1g-dev cmake ninja-build
+#Prepare dependency installation
+before_install:
+  - |
+    if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+      sudo apt-get -qq update
+    fi
 
-#install:
-#  - brew update
-#  - brew install ninja
+install:
+  - if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+      sudo apt-get -y -q --no-install-recommends install
+        zlib1g-dev libncursesw5-dev libgeoip-dev
+        nettle-dev libgmp-dev libcurl4-gnutls-dev libsdl2-dev
+        libogg-dev libvorbis-dev libopusfile-dev libtheora-dev
+        libwebp-dev libjpeg8-dev libpng-dev
+        libfreetype6-dev libglew-dev libopenal-dev
+        liblua5.2-dev;
+    fi
+  - |
+    if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+      sudo apt-get -y -q --no-install-recommends install ninja-build
+    elif [ "$TRAVIS_OS_NAME" == "osx" ]; then
+      brew install ninja
+    fi
+  - |
+    # workarounds to make ccache work
```

Comment:
I'm almost a totally newb there, what means that pipe?

Reply:
a literal multiline string; so the new lines from the yaml file are preserved

Here a great overview of this one and the other 8 types:
https://stackoverflow.com/questions/3790454/in-yaml-how-do-i-break-a-string-over-multiple-lines/21699210#21699210

Reply:
that's what my brain inferred, but in this case why is this pipe missing for the first `if` block in `install` target?

Reply:
ok so  the other way to not use pipe would be to put a `;` at the end of `sudo apt-get -y -q --no-install-recommends install ninja-build` and at the end of `brew install ninja`, right?

Reply:
In the apt-get command you don't want any new lines, so no pipe there; it will then be passed to the OS as a single line.

I.e.
```bash
apt-get install
something
```
would fail as not knowing the command `something`